watcher/watcher.go

```diff
@@ -606,27 +606,46 @@ func (w *Watcher) forwardRemovalRoomEvent(logger *logrus.Entry, rooms []string)
 	return nil
 }
 
-func (w *Watcher) listPodsToReplace(logger *logrus.Entry, podNames []string) (podsToReplace []v1.Pod, err error) {
+func (w *Watcher) listPods() (pods []v1.Pod, err error) {
 	k := w.KubernetesClient
-	pods, err := k.CoreV1().Pods(w.SchedulerName).List(
+	podList, err := k.CoreV1().Pods(w.SchedulerName).List(
 		metav1.ListOptions{})
 	if err != nil {
-		logger.WithError(err).Error("failed to list pods on namespace")
 		return nil, err
 	}
 
+	return podList.Items, nil
+}
+
+func (w *Watcher) listPodsToReplace(logger *logrus.Entry, pods []v1.Pod, podNames []string) (podsToReplace []v1.Pod) {
 	podNameMap := map[string]bool{}
 	for _, name := range podNames {
 		podNameMap[name] = true
 	}
 
-	for _, pod := range pods.Items {
+	for _, pod := range pods {
 		if podNameMap[pod.GetName()] {
 			podsToReplace = append(podsToReplace, pod)
 		}
 	}
 
-	return podsToReplace, nil
+	return podsToReplace
+}
+
+// zombie rooms are the ones that are in terminating state but the pods doesn't exist
+func (w *Watcher) removeZombies(pods []v1.Pod, rooms []string) {
+	zombieRooms := []*models.Room{}
+	liveKubePods := map[string]bool{}
+	for _, pod := range pods {
+		liveKubePods[pod.GetName()] = true
+	}
+	for _, room := range rooms {
+		if _, ok := liveKubePods[room]; !ok {
+			zombieRooms = append(zombieRooms, models.NewRoom(room, w.SchedulerName))
+		}
+	}
+
+	models.ClearAllMultipleRooms(w.RedisClient.Client, w.MetricsReporter, zombieRooms)
```

Comment:
why do you ignore the possible error here?

Reply:
I shouldn't. My bad